### PR TITLE
Added "clear", "undo" and "all" options

### PR DESCRIPTION
**Pull request by NRJulian**
Hi John! I've been playing with the note-taker app and added a few commands I thought I would like:
- "-c" clears all notes in the notes file
- "-a" pipes the notes file through less (a bit quicker for just checking than the -i/-e option)
-"-u" for undo, deletes the most recent note (deletes from the next-to-last line upwards till it reaches another empty line)
I also edited the README to include the added options, and I thought you might be interested in checking them out and maybe pulling them. Thank you!

### CODE REVIEW

**Comment by johncoleman83** [comment]
Hi, these are all great ideas and I'd love to add them.  I really like the `-a` option and this is very very handy instead of the option I have in place which is `-t` for tail.

Unfortunately, the `-u` option did not work for me.  The undo did not stop at the first blank line, and i do not know why yet.  Fortunately, I created a backup before doing this, which made me think that there should be an option to backup your notetaker file with some hash string.

I have not tried to debug this `-u` error that clears the entire file because it is difficult to see the changes in the change set since you did not start a new branch or reset your branch to the master  HEAD commit, so I see your entire commit history in this PR.  It is preferential for PR's that you either start a new branch off of `master` branch, OR with your current branch run 
```
$ git fetch origin master && git reset --hard origin/master
```
**NOTE:** you will lose your changes with this above command so you need to save them somehow

... and then start your changes.  If you can do that, I can help debug.  OR if you just want to start one option at a time that would be great to and you can start with the `-a` option.  Either way works, but if you can at least fix your commit history so that I can see only the unique changes in this PR that will help and I may just merge and fix the bug myself.

Thanks again for your contribution.

**Comment by johncoleman83** [comment]
@JiveCatDaddyO , in case you don't get notified, I'm using your handle here so hopefully you get ping'd

**Comment by johncoleman83** [comment]
Since I just merged updates from  @luisejrobles you can see how there are merge conflicts with all the old commit's lingering in this PR, so please definitely resolve those conflicts.  I also added to the conflicts by creating some updates.  I cleaned up a lot and added a backup option

**Comment by NRJulian** [comment]
Hi again! Thank you so much for the feedback!

>  The undo did not stop at the first blank line...

That is certainly not a good error to have! I originally tested the option with a fresh install using zsh in Arch...I will try some different shell/distro combinations to see if I can replicate and fix it.

I apologize for sending an inelegant PR: I am just learning the ins and outs of Github and your patience and comments are very helpful and appreciated! Since the files have changed considerably, would it be easier if I just pull the new version from your repo now, add/fix my features on a fork of that, and then send a new clean pull request? (Maybe one feature at a time as you suggested?)

Thank you!

**Comment by johncoleman83** [comment]
Hi @JiveCatDaddyO, no probs, and glad you still want to work on these updates.  I tested with `bash` and I think this should work with `bash` out of the box, since it is basic and `bash` is pretty common nowadays.

So, there are many ways to solve this issue of how to make a pull request or rather this can be simplified to merging 1 branch with another branch.

You can create a new branch (is that what you meant from fork?) from your master branch AND create a new PR or just use your same master branch and after you push your changes they should be recognized here so no need for a new PR.

You can pull but I think that may still create conflicts since I squashed all your commits (especially the ones with the exact same message) into 1 commit to represent your changes.  And it will likely create a new commit with the `merge from master` message that is a bit annoying.  So that's why I suggested that you just run: `git fetch && git reset --hard`.  I realize now that your origin is not this repository.  So you can set the upstream, then use the upstream (which is this repo) to reset your branch.  These commands should help, please ask if you don't understand:
```
$ git remote add upstream https://github.com/johncoleman83/note_taker
$ get remote -v # this is just to verify
$ git fetch upstream master
$ git reset --hard upstream/master
$ git checkout -b NEW_BRANCH_NAME # this is only optional if you want a new branch
$ git push --force origin/master # or instead of master use NEW_BRANCH_NAME
```
**NOTE:** You will lose your updates if you do this so be sure to save your updates somehow.  Also, if you create a new branch which I showed you in line 5 of the above commands then YES you will have to create a new PR.  If you chose to keep working on master, then NO, this PR should recognize any of your updates as long as your push them to your github repo, since this repo understands what is in your github repo.

**Comment by NRJulian** [comment]
This is tremendously helpful! Thank you!

**Comment by NRJulian** [comment]
Hello again @johncoleman83! I hope you are doing well!

I have followed your helpful guide, learned a lot about workflow, and hopefully synced up the repos. Note that I have not yet tried debugging the "-u" option: my first hurdle was to get over merge issues. However, '-u' did seem to work as expected with ```bash``` on my install of Ubuntu Server 16.04.3. What would you like me to do next?

Thanks!

File: note_taker.bash [hunk 1]
```diff
@@ -57,7 +57,65 @@ handle_options () {
...
+		((LINES--))
+		# get the text of the penultimate line and assign it to LINE_TEXT
+		LINE_NUMBER=$LINES
+		# get the text of the penultimate line and assign it to LINE_TEXT
+		LINE_TEXT=`tail -n +"$LINE_NUMBER" "$NOTE_PAD_PATH" | head -n 1`
+		# as long as LINE_TEXT is not empty and LINE_NUMBER is not 0
+		until [ -z "$LINE_TEXT" ] || [ "$LINE_NUMBER" -eq 0 ]; do
+			# delete the last line of the notes file by copying a shortened version
+			# to a temp file and then overwriting the original with the temp
+			head -n -1 "$NOTE_PAD_PATH" > temp.txt
```
Comment (johncoleman83) [inline note_taker.bash:103]: the bug is here:
on mac terminal bash, this `head -n -1 "$NOTE_PAD_PATH" > temp.txt` causes:
```
head: illegal line count -- -1
```
I'll merge this and make the change.
```
cat "$NOTE_PAD_PATH" | tail -r | tail -n +2 | tail -r > temp.txt
```

**Comment by johncoleman83** [comment]
Looks really good, works great, I'm gonna merge and make a quick update.  Thanks for the updates and for making this a clean commit!

File: note_taker.bash (hunk 1 above)
Comment (NRJulian) [reply note_taker.bash:103]: Nice fix!

**Comment by NRJulian** [comment]
Hooray! Thanks for all your help!

**Pull request merged**